Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -341,14 +348,20 @@ void TextureMapperLayer::computeTransformsRecursive(ComputeTransformData& data)
 #endif
 }
 
-void TextureMapperLayer::paint(TextureMapper& textureMapper)
+void TextureMapperLayer::prepareForPainting(TextureMapper& textureMapper)
 {
     processDescendantLayersFlatteningRequirements();
 
     ComputeTransformData data;
     computeTransformsRecursive(data);
     textureMapper.setDepthRange(data.zNear, data.zFar);
+}
 
+void TextureMapperLayer::paint(TextureMapper& textureMapper)
+{
+#if !ENABLE(DAMAGE_TRACKING)
+    prepareForPainting(textureMapper);
+#endif
```

Comment:
Damage tracking is also a runtime decision, shouldn't we do this only when using the damage? I think it's clearer if we handle this in the caller, because in CoordinatedGraphicsScene::paintToCurrentGLContext it's not obvious that is is called when not tracking damage.